plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+import { VERSION } from './version';
+import * as constants from './constants';
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationNodeModuleFile,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import * as utils from './utils';
+import * as types from './types';
+
+import * as knex from 'knex';
+
+const contextSymbol = Symbol('knexContextSymbol');
+const DEFAULT_CONFIG: types.Config = {
+  maxQueryLength: 1022,
+};
+
+export class KnexInstrumentation extends InstrumentationBase<typeof knex> {
+  constructor(config: types.Config = {}) {
+    super(
+      `@opentelemetry/instrumentation-${constants.MODULE_NAME}`,
+      VERSION,
+      Object.assign({}, DEFAULT_CONFIG, config)
+    );
+  }
+
+  private _moduleVersion?: string;
```

Comment:
If user has multiple installations of the same package with different versions, then since we have a single instance of the Instrumentation, they will both report the the same version (the last one being required).
I usually pass the module version from the `patch` callback directly to the patched function. 
But that's really a rare and minor issue.

Reply:
Good point. `InstrumentationBase` definition is kind of weird in that way. I'd love not to use `this` at all, but the fact that wrapper methods are on the instance of `InstrumentationBase` forces that. I do agree that's a rare case, but I think it should be fixed.

Issue for other cases I've done a similar solution: https://github.com/open-telemetry/opentelemetry-js-contrib/issues/516

Reply:
Do you have an alternative suggestion to the base class?

Reply:
I didn't intend to sound overly critical on that, because I really think it's generally fine. The current solution works, but there is also room for improvement.

If we are open to changing this, I could come up with a suggestion at some point. It's just that I've been busy with other things for now and I think that introducing an alternative interface for defining instrumentation could potentially live alongside or on top of the current version, so it hasn't been the priority for me.